utils/get_subscription.py

```diff
@@ -0,0 +1,82 @@
+from categories.models import Category,UserSubscription
+
+class get_subscription:
+    should_subscribe = {}
+    subscribe = []
+    unsubscribe = []
+    last_subscribed = {}
+    selected = {}
+    roots = []
+    
+    def __init__(self,save,drop,person,action):
+            self.save = save
+            self.drop = drop
+            self.person = person
+            self.all = save+drop
+            self.action = action
+            self.subscribe = []
+            self.unsubscribe = []
+            self.roots = []
+   
+    #nodes to be selected
+    def select(self):
+        for slug in self.save:
+            self.selected[slug] = True
+        for slug in self.drop:
+            self.selected[slug] = False
+
+    #gets all the roots in all categories
+    def get_roots(self):
+        for slug in self.all:
+            category = Category.objects.get(slug=slug)
```

Comment:
Might be better to reduce the number of db calls by using filter instead of get, using the slug names